src/components/AssetDid/AssetDid.tsx

```diff
@@ -1,116 +1,150 @@
+        <input
+          required
+          className={styles.didInputValue}
+          name="assetReference"
+          value={assetReference}
+          onChange={handleInput}
+        />
+      </label>
+      :
+      <label className={styles.didInputLabel}>
+        <p className={styles.tokenID}>
+          <span>Token ID</span>{' '}
+          <span className={styles.optional}>(Optional)</span>
+        </p>
+        <input
+          className={styles.didInputValue}
+          name="assetInstance"
+          value={assetInstance}
+          onChange={handleInput}
+        />
+      </label>
+    </section>
+  );
 }
 
-const initialValues = {
+const initialValues: AssetDidElements = {
   chainId: '',
   assetNamespace: '',
   assetReference: '',
   assetInstance: '',
 };
 
 export function AssetDid() {
-  const [assetDidInput, setAssetDidInput] = useState(initialValues);
-  const { chainId, assetNamespace, assetReference, assetInstance } =
-    assetDidInput;
+  const [didInput, setDidInput] = useState<AssetDidElements>(initialValues);
+  const { chainId, assetNamespace, assetReference } = didInput;
 
   const [assetDidUri, setAssetDidUri] = useState<AssetDidUri>();
 
-  const openSeaUrl = useMemo(() => {
-    if (!assetDidUri || !assetInstance) {
-      return;
-    }
-    return getOpenSeaUrl(assetDidUri);
-  }, [assetDidUri, assetInstance]);
+  const openSeaUrl = assetDidUri && getOpenSeaUrl(assetDidUri);
 
-  const urlInputError = useBooleanState();
-  const didInputError = useBooleanState();
+  const error = useBooleanState();
 
-  const handleUrlInput = useCallback(
-    (event: ChangeEvent<HTMLInputElement>) => {
-      event.preventDefault();
-      urlInputError.off();
-      didInputError.off();
-      setAssetDidUri(undefined);
-
-      const url = event.currentTarget.value;
-
-      if (isAssetDidUri(url)) {
-        const { chainId, assetNamespace, assetReference, assetInstance } =
-          parse(url);
-        setAssetDidInput({
-          chainId,
-          assetNamespace,
-          assetReference,
-          assetInstance: assetInstance || '',
-        });
-        return;
-      }
-
-      if (isOpenSeaUrl(url)) {
-        setAssetDidInput(parseOpenSeaUrl(url));
-        return;
-      }
-
-      urlInputError.on();
-      setAssetDidInput(initialValues);
-    },
-    [urlInputError, didInputError],
-  );
+  const reset = useCallback(() => {
+    error.off();
+    setDidInput(initialValues);
+    setAssetDidUri(undefined);
+  }, [error]);
 
   const handleDidInput = useCallback(
     (event: ChangeEvent<HTMLInputElement>) => {
-      event.preventDefault();
-      didInputError.off();
+      error.off();
       setAssetDidUri(undefined);
 
       const key = event.currentTarget.name;
       const value = event.currentTarget.value;
 
-      if (Object.keys(assetDidInput).includes(key)) {
-        setAssetDidInput({ ...assetDidInput, [key]: value });
+      if (Object.keys(didInput).includes(key)) {
```

Comment:
Would this work as well?
```suggestion
      if (key in didInput) {
```